packages/polaris-viz/src/components/Legend/Legend.tsx

```diff
@@ -36,6 +36,15 @@ export function Legend({
 }: LegendProps) {
   const {hiddenIndexes} = useExternalHideEvents();
 
+  const onDimensionChange = useCallback(
+    (index, dimensions: LegendItemDimension) => {
+      if (itemDimensions?.current) {
+        itemDimensions.current[index] = dimensions;
+      }
+    },
+    [itemDimensions],
+  );
+
```

Comment:
This isn't related to the bug, I just noticed that there would be a `Maximum stack depth has been reached` React error occasionally in LegendItem since `onDimensionChange` changed on every render 😅